gitlab/.gitlab-ci.yml

```diff
@@ -0,0 +1,417 @@
+    - openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} stage "$IMAGE_TO_BUILD"
+    # Run the deploy commands, append $PXE_VARS in case the pxe target is run manually
+    - echo "openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} deploy \"$IMAGE_TO_BUILD\" \"$PXE_VARS\""
+    - openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} deploy "$IMAGE_TO_BUILD" "$PXE_VARS"
+  after_script:
+    - BUILD_DIR=openxt-workspace/build-${CI_COMMIT_REF_NAME}
+    - if [[ "x$DEBUG" != "x" ]]; then ls -l $BUILD_DIR/deploy/*; fi
+    # Save the iso to its own directory if that was one of the targets
+    - |
+        if [[ ${IMAGE_TO_BUILD} == *"iso"* ]]; then
+          mkdir -p deploy/iso;
+          cp -f ${BUILD_DIR}/deploy/*.iso deploy/iso;
+        fi
+    # Preserve the contents of the deploy directory
+    - cp -Rf ${BUILD_DIR}/deploy/* deploy/
+    - if [[ "x$DEBUG" != "x" ]]; then ls -l deploy/*; fi
+  dependencies:
+    - setup:workspace
+    - build:dom0:image
+    - build:upgrade:image
+    - build:uivm:image
+    - build:ndvm:image
+    - build:syncvm:image
+  artifacts:
+    paths:
+      - deploy/iso
+    when: always
+    expire_in: 2 days
+
+stage:iso:
+  <<: *stage_definition
+  variables:
+    IMAGE_TO_BUILD: "iso"
+  only:
+    variables:
+      - $PACKAGE_TYPE == "iso"
+
+stage:pxe:
+  <<: *stage_definition
+  variables:
+    IMAGE_TO_BUILD: "pxe"
+  only:
+    variables:
+      - $PACKAGE_TYPE == "pxe"
+
+stage:all:
+  <<: *stage_definition
+  variables:
+    IMAGE_TO_BUILD: "iso pxe repository"
+  before_script: *setup_cm
+  dependencies:
+    - setup:workspace
+    - build:all
+  only:
+    variables:
+      - $PACKAGE_TYPE == "all"
+
+.deploy_template: &deploy_definition  # Template for saving the iso to S3
+  image: alpine:latest
+  stage: deploy
+  before_script:
+    # Install the coreutils package and awscli python package, this enables easy upload to S3
+    - apk -q update && apk -qUu add py-pip && pip install --quiet awscli
+  script:
+    - if [[ "x$DEBUG" != "x" ]]; then echo "ls -al deploy"; ls -lR deploy/; fi
+    # Upload the iso to a directory named after today and the branch. Make it expire in $NUM_DAYS_TO_EXPIRE days, defaults to 30
+    - |
+      if [ -d deploy/iso ]; then
+        let EXPIRE_HOURS=$NUM_DAYS_TO_EXPIRE*24;
+        EXPIRATION_DATE=`date -d "$EXPIRE_HOURS:00:00" -I`;
+        TODAY=`date -I`;
+        aws s3 cp deploy/iso/* s3://${S3_BUCKET}/iso/$TODAY/$CI_COMMIT_REF_NAME/ --sse-kms-key-id $AWS_S3_ISO_KEY_ID --sse aws:kms --expires $EXPIRATION_DATE;
+      fi
+
+deploy:iso:
+  <<: *deploy_definition
+  dependencies:
+    - setup:workspace
+    - stage:iso
+  only:
+    variables:
+      - $PACKAGE_TYPE == "iso"
+
+deploy:all:
+  <<: *deploy_definition
+  dependencies:
+    - setup:workspace
+    - stage:all
+  only:
+    variables:
+      - $PACKAGE_TYPE == "all"
+
+cleanup:sstate:kernel:
+  image: debian:latest
+  stage: prepare
+  script:
+    # Delete anything that has not been accessed in 4 days
```

Comment:
Copy/paste typo with `cleanup:sstate:prune:`, it deletes all kernel/out-of-tree modules.